### PR TITLE
Refactor SAML

### PR DESCRIPTION
**Pull request by mrajashree**
1. Remove use of SAML library's middleware, so that we don't use the token that
the library sets. This token set by the library can be big depending on size of
user's assertion returned by IdP. This token is set as a cookie and big cookies
cause problems as most browsers have a max limit of cookie size of 4096 bytes
2. Make the auth flow similar to 2.0
3. Make sure token cookie's secure flag is set depending on the request
rancherlabs/the-ranch#151
4. Restrict redirectBackBase to whitelisted domains (Host registration URL)
rancherlabs/the-ranch#150

https://github.com/rancher/rancher/issues/14756

rancherlabs/the-ranch#149

### CODE REVIEW

File: service/route_handlers.go [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
 
 const (
-	samlParam        = "samlJWT"
 	redirectBackBase = "redirectBackBase"
 	redirectBackPath = "redirectBackPath"
-	getSamlAuthToken = "/v1-auth/saml/authtoken"
 )
 
+var redirectBackBaseValue, redirectBackPathValue string
+var XForwardedProto string
```
Comment (cjellick) [inline service/route_handlers.go]: bad to have these as global vars

Comment (mrajashree) [reply service/route_handlers.go]: @cjellick okay yes, I can add them to the clientState

File: providers/shibboleth/shibboleth_client.go [hunk 2]
```diff
@@ -23,52 +18,33 @@ import (
...
-	var idpURL string
 	var privKey *rsa.PrivateKey
 	var cert *x509.Certificate
 	var err error
 	var ok bool
 
-	sp.config = configToSet
+	spclient.config = configToSet
 
-	if configToSet.IDPMetadataURL == "" {
```
Comment (cjellick) [inline providers/shibboleth/shibboleth_client.go]: All this theoretically dead code you removed to simplify configuration - can you NOT remove it?
There is a chance that in v1.6, someone is actually using IDPMetadataFilePath or SPSelfSignedCertFilePath or SPSelfSignedKeyFilePath

**Comment by cjellick** [comment]
Looks good outside of the two comments that I've made. So, once those get addressed, we should be good to go.

File: providers/shibboleth/shibboleth_client.go (hunk 2 above)
Comment (mrajashree) [reply providers/shibboleth/shibboleth_client.go]: @cjellick I remember checking that file paths weren't working, cattle code was missing something. I'll check what it was
Update: Oh right, user has no way of providing them in cattle!

**Comment by cjellick** [comment]
LGTM

**Comment by mrajashree** [comment]
https://github.com/rancherlabs/the-ranch/issues/149
https://github.com/rancherlabs/the-ranch/issues/150
https://github.com/rancherlabs/the-ranch/issues/151

**Pull request merged**